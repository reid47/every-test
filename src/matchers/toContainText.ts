import { formatValue } from './common';

export function toContainText(received, expectedText) {
  const actualText = received.textContent;

  const pass =
    typeof expectedText.test === 'function'
      ? expectedText.test(actualText)
      : actualText.indexOf(expectedText) > -1;

  const message = pass
    ? `Expected element not to contain text ${formatValue(expectedText)}.`
    : `Expected element to contain text ${formatValue(
        expectedText
      )}, but actual text was ${formatValue(actualText)}.`;

  return { pass, message };
}
