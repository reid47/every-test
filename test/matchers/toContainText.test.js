import toContainText from '../../src/matchers/toContainText';

describe('toContainText', () => {
  test('passing (substring)', () => {
    const el = document.createElement('div');
    const text = document.createTextNode('hello');
    el.appendChild(text);

    const result = toContainText(el, 'h');
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('passing (regex)', () => {
    const el = document.createElement('div');
    const text = document.createTextNode('hello');
    el.appendChild(text);

    const result = toContainText(el, /ell/);
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing', () => {
    const el = document.createElement('div');
    const text = document.createTextNode('hello');
    el.appendChild(text);

    const result = toContainText(el, 'hello!');
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });
});
