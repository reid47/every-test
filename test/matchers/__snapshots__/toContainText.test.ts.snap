// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toContainText failing 1`] = `"Expected element to contain text \\"hello!\\", but actual text was \\"hello\\"."`;

exports[`toContainText passing (regex) 1`] = `"Expected element not to contain text {}."`;

exports[`toContainText passing (substring) 1`] = `"Expected element not to contain text \\"h\\"."`;
